src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -191,7 +192,27 @@ export class EthereumAdapter extends BaseProtocolAdapter {
   }
 
   override async fetchAvailableTokens(): Promise<IToken[]> {
-    return [DUMMY_ERC20_TOKEN]; // TODO replace with real tokens fetching
+    try {
+      // Temporary solution for fetching the ERC-20 tokens.
+      // TODO Replace with our own node API
+      const tokens = (await fetchJson('https://api.ethplorer.io/getTop?apiKey=freekey'))?.tokens || [];
+
+      return tokens
+        .filter(({ address, name, symbol }: any) => (address && name && symbol))
+        // Remove the ETH coin from the results because as we are adding it manually
+        .filter(({ address }: any) => address !== ETH_CONTRACT_ID_EXTERNAL)
+        .map((token: any): IToken => ({
+          contractId: token.address,
+          decimals: 0, // TODO api does not provide this value so we need to come up how to get this
```

Comment:
@CedrikNikita we discussed this. Did you find any solution for this?

Reply:
OK, I'll updated this in this PR: https://github.com/superhero-com/superhero-wallet/pull/2594/files#diff-a9ba9cbedd19c9f66d564d2e89912890209b98f0a7ef19187877d2587300e476R162